DEV: (cmds) add TS.QUERYLABELS command page (#3633)
redis/docs: content/commands/ts.querylabels.md

---
acl_categories:
- '@read'
- '@timeseries'
arguments:
- arguments:
  - display_text: LABELS
    name: LABELS
    token: LABELS
    type: pure-token
  - arguments:
    - display_text: VALUES
      name: VALUES
      token: VALUES
      type: pure-token
    - display_text: label
      name: label
      type: string
    name: VALUES
    type: block
  name: subtype
  type: oneof
- arguments:
  - display_text: FILTER
    name: FILTER
    token: FILTER
    type: pure-token
  - display_text: filterExpr
    multiple: true
    name: filterExpr
    type: string
  name: FILTER
  optional: true
  type: block
arity: -2
categories:
- docs
- develop
- stack
- oss
- rs
- rc
- oss
- kubernetes
- clients
command_flags:
- readonly
- module
complexity: O(n) where n is the number of time-series that match the filters (all
  indexed series when FILTER is omitted)
description: Get all label names, or all values of a given label, for time series
  matching a filter list, or all series

> group: module

Reviewer: Should this be
```suggestion
group: timeseries
```
...rather than `module`? It doesn't seem to show the other time series commands in the right hand sidebar.

Author: Fixed in next push.

</details>

## Optional arguments

<details open>
<summary><code>FILTER filterExpr [filterExpr ...]</code></summary>

restricts the command to the time series whose labels and label values match all of the filter expressions. When `FILTER` is omitted, the command considers all indexed time series.

Each filter expression has one of the following syntaxes:

  - `label!=` - the time series has a label named `label`
  - `label=value` - the time series has a label named `label` with a value equal to `value`
  - `label=(value1,value2,...)` - the time series has a label named `label` with a value equal to one of the values in the list
  - `label=` - the time series does not have a label named `label`
  - `label!=value` - the time series does not have a label named `label` with a value equal to `value`
  - `label!=(value1,value2,...)` - the time series does not have a label named `label` with a value equal to any of the values in the list

  <note><b>Notes:</b>
   - At least one filter expression with a syntax `label=value` or `label=(value1,value2,...)` is required.

>    - Filter expressions are conjunctive. For example, the filter `type=temperature room=study` means that a time series is a temperature time series of a study room.

Reviewer: I'm not sure if "means that" tends to imply that these particular label-value pairs are part of an existing ontology for labels. Maybe it's clearer to say "...means `type=temperature` AND `room=study`"? Then again maybe it's just me :-)

Author: Thank you. Fixed in next push.

>    - Whitespaces are unallowed in a filter expression except between quotes or double quotes in values - e.g., `x="y y"` or `x='(y y,z z)'`.

Reviewer: Don't think it's normally used in the plural? And I do believe we have an "e.g." in there!
```suggestion
   - Whitespace is not allowed in a filter expression except between quotes or double quotes in values - for example, `x="y y"` or `x='(y y,z z)'`.
```

Author: Better living through chemistry. 👀

Fixed in next push.

  <note><b>Notes:</b>
   - At least one filter expression with a syntax `label=value` or `label=(value1,value2,...)` is required.
   - Filter expressions are conjunctive. For example, the filter `type=temperature room=study` means that `type=temperature` AND `room=study`.

>    - Whitespace is not allowed in a filter expression except between quotes or double quotes in values - for example, `x="y y"` or `x='(y y,z z)'`.

Reviewer 2: Please fix the other filter commands as well.

Author: Done. I made the text an "embed" (that's Hugo's method for de-duplication).
